src/classes/GRER_SyncGradeLevel_TDTM.cls

```diff
@@ -90,16 +93,29 @@ global with sharing class GRER_SyncGradeLevel_TDTM extends hed.TDTM_Runnable {
                         contactsToUpdate.add(relatedContact);
                     } else {
                         gradeErol.Status__c = 'Former';
-                        gradeEnrolsToUpdate.add(gradeErol);    
+                        gradeEnrolsToUpdate.add(gradeErol);
                     }
                 }
-            }  
+            }
+            if (triggerAction == hed.TDTM_Runnable.Action.AfterInsert) {
+                TDTM_ProcessControl.setRecursionFlag(TDTM_ProcessControl.registeredTrigger.GRER_SyncGradeLevel_TDTM_After_Insert, true);
+            }
+            if (triggerAction == hed.TDTM_Runnable.Action.AfterUpdate) {
+                TDTM_ProcessControl.setRecursionFlag(TDTM_ProcessControl.registeredTrigger.GRER_SyncGradeLevel_TDTM_After_Update, true);
+            }
         }
 
         //Add toupdate lists to dmlWrapper
         dmlWrapper.objectsToUpdate.addAll((List<SObject>)contactsToUpdate);
         dmlWrapper.objectsToUpdate.addAll((List<SObject>)gradeEnrolsToUpdate);
 
+        TDTM_ProcessControl.setRecursionFlag(TDTM_ProcessControl.registeredTrigger.GRER_SyncGradeLevel_TDTM_After_Insert, false);
```

Comment:
Can we leave a comment here on why these flags are being manipulated?

Reply:
Added comments!